models/rocifi/polygon/rocifi_polygon_v2_loan_master.sql

```diff
@@ -0,0 +1,302 @@
+{{config (
+    alias = 'v2_loan_master',
+    post_hook = '{{ 
+        expose_spells(\'["polygon"]\',
+        "project", 
+        "rocifi",
+        \'["maybeyonas"]\') }}'
+)}} with score as (
+    select evt_block_time,
+        score,
+        tokenId as token_id
+    from {{source('rocifi_v2_polygon', 'ScoreDB_evt_ScoreUpdated')}}
+),
+NFCS as (
+    select m.evt_block_time as mint_time,
+        _recipient as minter,
+        _tokenId::int as token_id,
+        explode(_addressBundle) as addr
+    from {{source('rocifi_polygon', 'NFCS_evt_TokenMinted')}} m
+),
+loan_id_score as (
+    -- finding the credit score during the loanCreation, by finding the last 'scoreUpdate' before the Loan is created
+    select loan_id,
+        coalesce(score, 404) as nfcs_score,
+        token_id as nfcs_id
+    from (
+            select evt_block_time,
+                event,
+                loan_id,
+                coalesce(
+                    score,
+                    last(score, true) over(
+                        partition by token_id
+                        order by evt_block_time range unbounded preceding
+                    )
+                ) as score,
+                token_id
+            from (
+                    select evt_block_time,
+                        'updateScore' as event,
+                        score,
+                        token_id,
+                        null as loan_id
+                    from score
+                    union all
+                    select evt_block_time,
+                        'loanCreated' as event,
+                        null as score,
+                        token_id,
+                        loanId as loan_id
+                    from {{source(
+                            'rocifi_v2_polygon',
+                            'LoanManager_evt_LoanCreated'
```

Comment:
We can rewrite the nested sql in CTE structure.
Also we may modify the model to incremental view.

Reply:
@Hosuke I have updated to CTE structure and remove Pipfile changes. 
Can we stick to the `view` type for now. I was thinking of shifting to a `incremental` model down the line, as the Table is kinda an urgent requirement.